litestar/_openapi/responses.py

```diff
@@ -240,14 +241,20 @@ def create_additional_responses(self) -> Iterator[tuple[str, OpenAPIResponse]]:
                 prefer_alias=False,
                 generate_examples=additional_response.generate_examples,
             )
-            schema = schema_creator.for_field_definition(
-                FieldDefinition.from_annotation(additional_response.data_container)
-            )
+            field_def = FieldDefinition.from_annotation(additional_response.data_container)
+            schema = schema_creator.for_field_definition(field_def)
+
+            examples: dict[str, Example | Reference] | None
+            if additional_response.examples:
+                examples = dict(get_formatted_examples(field_def, additional_response.examples))
```

Comment:
`OpenAPIMediaType.examples` declares `dict[str, ...]` but `get_formatted_examples()` returns `Mapping`, so had to convert to dict.

Alternatively, should I adjust `OpenAPIMediaType.examples` to accept `Mapping` instead?

Reply:
> Alternatively, should I adjust OpenAPIMediaType.examples to accept Mapping instead?

I think that's fine, yes